From b879f4476882214b1cb75b6ec9d36baf746a7bee Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:52 -0700
Subject: [PATCH 0566/1852] dmaengine: xilinx: frmbuf: Make max-width and
 max-height mandatory properties

commit 38e75ec40dffbd5771ca7b0a8a3df53a4d7aa1a3 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes the xlnx,max-width and xlnx,max-height mandatory
properties in the device tree node. Since this breaks backward
compatibility, a warning message is added.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index 42c9c6d58bd4..a135af17129b 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -1538,7 +1538,8 @@ static int xilinx_frmbuf_probe(struct platform_device *pdev)
 
 	err = of_property_read_u32(node, "xlnx,max-height", &xdev->max_height);
 	if (err < 0) {
-		xdev->max_height = XILINX_FRMBUF_MAX_HEIGHT;
+		dev_err(xdev->dev, "xlnx,max-height is missing!");
+		return -EINVAL;
 	} else if (xdev->max_height > XILINX_FRMBUF_MAX_HEIGHT ||
 		   xdev->max_height < XILINX_FRMBUF_MIN_HEIGHT) {
 		dev_err(&pdev->dev, "Invalid height in dt");
@@ -1547,7 +1548,8 @@ static int xilinx_frmbuf_probe(struct platform_device *pdev)
 
 	err = of_property_read_u32(node, "xlnx,max-width", &xdev->max_width);
 	if (err < 0) {
-		xdev->max_width = XILINX_FRMBUF_MAX_WIDTH;
+		dev_err(xdev->dev, "xlnx,max-width is missing!");
+		return -EINVAL;
 	} else if (xdev->max_width > XILINX_FRMBUF_MAX_WIDTH ||
 		   xdev->max_width < XILINX_FRMBUF_MIN_WIDTH) {
 		dev_err(&pdev->dev, "Invalid width in dt");
-- 
2.31.1

